manifests/init.pp

```diff
@@ -99,7 +99,7 @@
     file { "${log_dir}/${file}":
       ensure => $directory_ensure,
       force  => true,
-      mode   => '0755',
+      mode   => '0777', # FIXME: this is a hack
```

Comment:
the user inside the container does not map to `root` on the host, so it can't write here :/

Reply:
Why do we even run it as root? Shouldn't it be able to run as some non-root user? Isn't that also better for the whole security aspect?

Reply:
Which `we` and `it`? :)

* Puppet (we?) starts the service/container as root, but Podman defaults to use user namespacing by default, so the root in the container that gets launched is not the root on the host.
* The user inside the container (it?) is not root anyway, but `1001`: https://github.com/RedHatInsights/iop-advisor-engine/blob/main/Containerfile#L2, which again is not `1001` on the host (or at least, should not be).

Reply:
> Puppet (we?) starts the service/container as root, but Podman defaults to use user namespacing by default, so the root in the container that gets launched is not the root on the host.
> The user inside the container (it?) is not root anyway, but `1001`: https://github.com/RedHatInsights/iop-advisor-engine/blob/main/Containerfile#L2, which again is not `1001` on the host (or at least, should not be).

That's indeed the `we` and `it` I intended.

I was wondering if the user namespacing wasn't causing the problems. https://docs.podman.io/en/latest/markdown/podman-systemd.unit.5.html#user says it can map to a real UID on the host and I'd think the file permission issue is resolved then.

Reply:
Are you suggesting to do something like `User=foreman` or `User=foreman-iop-whatever-yada-yada` and let the folder be owned by that user?
(No idea if that'd work with the `1001` baked into the container, but we can try)

Another aspect: at least for the logging part, this will probably go away, as we discussed to "just log to stdout" so that "it lands in the journal instead of a new file nobody knows about yet"

Reply:
Just tried `User=foreman` and that gives me:
```
Error: unable to find user foreman: no matching entries in passwd file
```

Which makes sense, there is no `foreman` inside the container.

Using `User = 993` (the UID of `foreman` in my deployment) seems to work.

Reply:
> Are you suggesting to do something like `User=foreman` or `User=foreman-iop-whatever-yada-yada` and let the folder be owned by that user?

Pretty much yes. https://docs.podman.io/en/latest/markdown/podman-systemd.unit.5.html#userns could help:
```ini
UserNS=keep-id:uid=1001
```

Reply:
I think a proper logging setup is a good action item to follow up on.